project-management/src/main/java/life/qbic/projectmanagement/application/api/AsyncProjectService.java

```diff
@@ -355,6 +421,45 @@ record ProjectCreationRequest(ProjectDesign design, ProjectContacts contacts,
 
   }
 
+  /**
+   * A service request to create one or more new samples for a project.
+   *
+   * @param projectId the project ID of the project the samples shall be created for
+   * @param requests  a collection of {@link SampleRegistrationRequest} items
+   * @since 1.10.0
+   */
+  record SampleCreationRequest(String projectId, Collection<SampleRegistrationRequest> requests) {
+    public SampleCreationRequest(String projectId, Collection<SampleRegistrationRequest> requests) {
+      this.projectId = projectId;
+      this.requests = List.copyOf(requests);
+    }
+  }
+
+  /**
+   * A service request to update one or more samples in a project.
+   *
+   * @param projectId the project ID of the project the samples shall be updated in
+   * @param requests  a collection for {@link SampleUpdate} items
+   * @since 1.10.0
+   */
+  record SampleUpdateRequest(String projectId, Collection<SampleUpdate> requests) {
```

Comment:
You probably want to overwrite the getter for the requests as well to return an unmodifiable list.